packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -601,7 +601,7 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
 
   int? search(List<DropdownMenuEntry<T>> entries, TextEditingController textEditingController) {
     final String searchText = textEditingController.value.text.toLowerCase();
-    if (searchText.isEmpty) {
+    if (searchText.isEmpty || entries.isEmpty) {
       return null;
     }
     if (currentHighlight != null && entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
```

Comment:
It is still possible to have out of range error when `entries` reduce in size but still non-empty.
For example, by highlighting the second item, then enter a search that only matches a single item would still cause out of range error in `entries[currentHighlight!]`.